csi.proto

```diff
@@ -1296,6 +1345,55 @@ message VolumeUsage {
   // Units by which values are measured. This field is REQUIRED.
   Unit unit = 4;
 }
+
+message VolumeHealth {
+  option (alpha_message) = true;
+
+  enum Condition {
+    // Volume health condition is unknown.
+    UNKNOWN = 0;
+    // Volume is accessible and no problems are detected. 
+    HEALTHY = 1;
+    // Volume is not accessible currently, but there's no reason to
+    // expect data loss. Examples: power outage, network outage,
+    // maintenance.
+    TEMPORARY_INACCESSIBLE = 10;
+    // Volume is accessible currently, but is degraded. Data loss
+    // is unlikely as the storage system may be rebuilding to fix the
+    // problem automatically. For example, a replica is lost on
+    // the storage system but a spare is configured, and the storage
+    // system is resyncing/reconstructing.
+    TEMPORARY_DEGRADED = 20;
+    // Volume is accessible currently, but a problem is detected that
+    // could lead to eventual data loss. Examples: recoverable hardware
+    // failure (i.e. the disk needs to be manually replaced by a human),
+    // cooling failure, equipment significantly past end of life.
+    FAILURE_LIKELY = 30;
+    // Volume is not accessible currently, and there is reason to
+    // believe the outage may be permanent. Examples: unrecoverable
+    // hardware failure, natural disasters, damage to equipment.
+    FATAL = 40;
+  }
+
+  message VolumeHealthError {
+    // The code describing the health condition of the volume.
+    // This is an opaque field to CO.
+    // This field is OPTIONAL.
+    string code = 1;
+
+    // The error message associated with the above error code.
+    // This field is REQUIRED.
+    string message = 2;
+  }
+
+  // The level of the health condition of the volume.
+  // This field is REQUIRED.
+  Condition condition = 1;
```

Comment:
Given how much disagreement there was about error codes on the calls ([Notes](https://docs.google.com/document/d/1-oiNg5V_GtS_JBAEViVBhZ3BYVFlbSz70hreyaD7c5Y/edit#heading=h.f7yywna1ikc7)) and the fact that on the Kubernetes side we intend to only expose this as FYI info for user (not for programmatic consumption), let's scope this down to just opaque message, and we can add in error codes in the future once we have consensus.

Reply:
Hi Saad, there were some disagreements during the 2/11 meeting, but I thought those got resolved during the review process.  In the last meeting on 3/4, no one raised any concerns about this PR either.  Should we schedule an adhoc meeting to discuss about the "Condition" part and see if we can finalize this?  Thanks!

Reply:
My understanding of the 2/11 meeting was that there was a lot of disagreement about the error codes. I'm sorry I was unable to attend the 3/4 meeting. I'm ok with doing an adhoc meeting to discuss further. My stance is less is more -- given that the initial goal of this design are that this is *not* intended to be consumed programatically, we should be very careful adding in fields that are clearly designed for programatic use.

Reply:
Yes, there were disagreements about the error codes (I’ve removed them to address your comments).  Folks like Conditions (or severity level) better.  I think we have come a long way on reaching some agreement on the Conditions thru code reviews from multiple people.  Yes, this is not intended to be used programmatically in the initial phase, but some kind of Condition can tell user quickly how bad the situation is and decide whether to dive into the detailed messages immediately or wait until later.  So I think it is very helpful to have the Condition in addition to the messages.

Reply:
Addressed the comments based on what we decided at the review meetings.